services/gundeck/src/Gundeck/Push.hs

```diff
@@ -172,8 +188,15 @@ pushAll pushes = do
         resp <- compilePushResps targets <$> mpaBulkPush (compilePushReq <$> targets)
 
         -- native push
-        forM_ resp $ \((notif, psh), alreadySent) -> unless (psh ^. pushTransient) $
-            mpaPushNative notif psh =<< nativeTargets psh alreadySent
+        forM_ resp $ \((notif :: Notification, psh :: Push), alreadySent :: [Presence]) -> do
```

Comment:
At this point, `alreadySent == number of devices of all users in a conversation, right?`?

Reply:
No, `alreadySent` contains what went out over websocket and doesn't need to be pushed natively.  I think conversations aren't really a thing here, we just have the `Notification` and `Push` to determine who should hear from this.